CMSgov/design-system
# [WNMGDS-3028] Updated Angular Example

## Summary

A substantial number of changes have been made to enable our custom, preactement-based web component build process, to generate web components that can function when nested inside of other web components, or framework components like those used in Angular. This particular pull request is both proof that things are working (yay!) and a test to see that our web components could be used in a production-like manner. 

The examples here are drawn from our existing Angular example but augmented to allow wrapping a la discussions we've had with developers on the CCXP team. The changes here are designed to address concerns by demonstrating example patterns. Specifically we address:
1. Setting attributes via inputs passed from parent components
2. Creating re-usable wrappers for our web components 
3. Monitoring and using our custom events in a couple of different ways
4. Demonstrating that nested web components can alter global state via Outputs
5. Unwrapped components work out of the box
6. Analytics events can be listened for, though I have not gotten them to work on ds-dialog or ds-drawer components. Not sure if it's my Angular-fu (minimal) or my analytics-fu (also minimal)
7. Importing styling and scripts globally via the `angular.json` file

#### A thing I need to do yet is update the README.md with some guidance on usage patterns. 

### What feedback am I looking for here? (listed in descending order of importance)
1. Are these examples demonstrative of things we want Angular/web component teams to know? (This question is one that will be asked of CCXP.)
2. Do some of the patterns I used here make sense? (This question is one that will be asked of CCXP.)
3. What's going on with the Anayltics events on the Dialog and Drawer?
5. Should I include other components?
6. Should I wrap all the components?
7. Does this look like trash? (styling updates maybe?)

## How to test

1. Pull down branch
9. Build as per usual
10. Navigate to `/examples/angular`
11. `yarn build`
12. `yarn start`
13. Go to `localhost:4200`
14. Observe! Click around. Look in the console! Take time to think.

## Checklist
